src/Native/Unix/System.Native/pal_networkstatistics.c

```diff
@@ -163,31 +171,28 @@ int32_t SystemNative_GetIcmpv4GlobalStatistics(Icmpv4GlobalStatistics* retStats)
-    retStats->DestinationUnreachableMessagesReceived = inHist[ICMP_UNREACH];
-    retStats->DestinationUnreachableMessagesSent = outHist[ICMP_UNREACH];
-    retStats->EchoRepliesReceived = inHist[ICMP_ECHOREPLY];
-    retStats->EchoRepliesSent = outHist[ICMP_ECHOREPLY];
-    retStats->EchoRequestsReceived = inHist[ICMP_ECHO];
-    retStats->EchoRequestsSent = outHist[ICMP_ECHO];
-    retStats->ParameterProblemsReceived = inHist[ICMP_PARAMPROB];
-    retStats->ParameterProblemsSent = outHist[ICMP_PARAMPROB];
-    retStats->RedirectsReceived = inHist[ICMP_REDIRECT];
-    retStats->RedirectsSent = outHist[ICMP_REDIRECT];
-    retStats->SourceQuenchesReceived = inHist[ICMP_SOURCEQUENCH];
-    retStats->SourceQuenchesSent = outHist[ICMP_SOURCEQUENCH];
-    retStats->TimeExceededMessagesReceived = inHist[ICMP_TIMXCEED];
-    retStats->TimeExceededMessagesSent = outHist[ICMP_TIMXCEED];
-    retStats->TimestampRepliesReceived = inHist[ICMP_TSTAMPREPLY];
-    retStats->TimestampRepliesSent = outHist[ICMP_TSTAMPREPLY];
-    retStats->TimestampRequestsReceived = inHist[ICMP_TSTAMP];
-    retStats->TimestampRequestsSent = outHist[ICMP_TSTAMP];
+    retStats->AddressMaskRepliesReceived = systemStats.icps_inhist[ICMP_MASKREPLY];
+    retStats->AddressMaskRepliesSent = systemStats.icps_outhist[ICMP_MASKREPLY];
+    retStats->AddressMaskRequestsReceived = systemStats.icps_inhist[ICMP_MASKREQ];
+    retStats->AddressMaskRequestsSent = systemStats.icps_outhist[ICMP_MASKREQ];
+    retStats->DestinationUnreachableMessagesReceived = systemStats.icps_inhist[ICMP_UNREACH];
+    retStats->DestinationUnreachableMessagesSent = systemStats.icps_outhist[ICMP_UNREACH];
+    retStats->EchoRepliesReceived = systemStats.icps_inhist[ICMP_ECHOREPLY];
+    retStats->EchoRepliesSent = systemStats.icps_outhist[ICMP_ECHOREPLY];
+    retStats->EchoRequestsReceived = systemStats.icps_inhist[ICMP_ECHO];
+    retStats->EchoRequestsSent = systemStats.icps_outhist[ICMP_ECHO];
+    retStats->ParameterProblemsReceived = systemStats.icps_inhist[ICMP_PARAMPROB];
+    retStats->ParameterProblemsSent = systemStats.icps_outhist[ICMP_PARAMPROB];
+    retStats->RedirectsReceived = systemStats.icps_inhist[ICMP_REDIRECT];
+    retStats->RedirectsSent = systemStats.icps_outhist[ICMP_REDIRECT];
+    retStats->SourceQuenchesReceived = systemStats.icps_inhist[ICMP_SOURCEQUENCH];
+    retStats->SourceQuenchesSent = systemStats.icps_outhist[ICMP_SOURCEQUENCH];
+    retStats->TimeExceededMessagesReceived = systemStats.icps_inhist[ICMP_TIMXCEED];
+    retStats->TimeExceededMessagesSent = systemStats.icps_outhist[ICMP_TIMXCEED];
+    retStats->TimestampRepliesReceived = systemStats.icps_inhist[ICMP_TSTAMPREPLY];
+    retStats->TimestampRepliesSent = systemStats.icps_outhist[ICMP_TSTAMPREPLY];
+    retStats->TimestampRequestsReceived = systemStats.icps_inhist[ICMP_TSTAMP];
+    retStats->TimestampRequestsSent = systemStats.icps_outhist[ICMP_TSTAMP];
```

Comment:
Why did this change?

Reply:
The icps_ is array of uint32_t on MAcOS and unsigned long on nFreeBSD. Compiler complained about pointer to int of different size. My initial code looked like
```c 
#if defined(__FreeBSD__)
unsigned long * inHist = systemStats.icps_inhist;		     
unsigned long * outHist = systemStats.icps_outhist;
#else
u_int32_t* inHist = systemStats.icps_inhist;
u_int32_t* outHist = systemStats.icps_outhist;
#endif
```
but I did not like it as we need to choose explicitly. New code is bigger change but it just works as compilers knows the types.

Reply:
Ok

Reply:
You could also use the following:
```
__typeof(systemStats.icps_inhist) inHist = systemStats.icps_inhist;
__typeof(systemStats.icps_outhist) outHist = systemStats.icps_outhist;
```
and keep the rest of the stuff untouched.